x-pack/plugins/fleet/public/applications/fleet/components/form_settings/index.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z, ZodFirstPartyTypeKind } from 'zod';
+import React, { useState } from 'react';
+import {
+  EuiDescribedFormGroup,
+  EuiFieldNumber,
+  EuiFieldText,
+  EuiFormRow,
+  EuiLink,
+} from '@elastic/eui';
+
+import type { SettingsConfig } from '../../../../../common/settings/types';
+import { useAgentPolicyFormContext } from '../../sections/agent_policy/components/agent_policy_form';
+
+export const settingComponentRegistry = new Map<
+  string,
+  (settingsconfig: SettingsConfig) => React.ReactElement
+>();
+
+settingComponentRegistry.set(z.number()._def.typeName, (settingsConfig) => {
+  return (
+    <SettingsFieldWrapper
+      settingsConfig={settingsConfig}
+      typeName={z.number()._def.typeName}
+      renderItem={({ fieldKey, fieldValue, handleChange, isInvalid, coercedSchema }: any) => (
+        <EuiFieldNumber
+          fullWidth
+          data-test-subj={fieldKey}
+          value={fieldValue}
+          onChange={handleChange}
+          isInvalid={isInvalid}
+          min={coercedSchema.minValue ?? undefined}
+          max={coercedSchema.maxValue ?? undefined}
+        />
+      )}
+    />
+  );
+});
+
+settingComponentRegistry.set(z.string()._def.typeName, (settingsConfig) => {
+  return (
+    <SettingsFieldWrapper
+      settingsConfig={settingsConfig}
+      typeName={z.string()._def.typeName}
+      renderItem={({ fieldKey, fieldValue, handleChange, isInvalid }: any) => (
+        <EuiFieldText
+          fullWidth
+          data-test-subj={fieldKey}
+          value={fieldValue}
+          onChange={handleChange}
+          isInvalid={isInvalid}
+        />
+      )}
+    />
+  );
+});
+
+const SettingsFieldWrapper: React.FC<{
+  settingsConfig: SettingsConfig;
+  typeName: keyof typeof ZodFirstPartyTypeKind;
+  renderItem: Function;
+}> = ({ settingsConfig, typeName, renderItem }) => {
+  const [error, setError] = useState('');
+  const agentPolicyFormContext = useAgentPolicyFormContext();
+
+  const fieldKey = `configuredSetting-${settingsConfig.name}`;
+  const defaultValue: number =
+    settingsConfig.schema instanceof z.ZodDefault
+      ? settingsConfig.schema._def.defaultValue()
+      : undefined;
+  const coercedSchema = settingsConfig.schema as z.ZodString;
+
+  const convertValue = (value: string, type: keyof typeof ZodFirstPartyTypeKind): any => {
+    if (type === ZodFirstPartyTypeKind.ZodNumber) {
```

Comment:
this part is different for number vs string type: for number we have to parse the updated value coming from the number field, as it is string